### ISSUE TITLE
Build failure with x509-store-1.6.4 using x509-1.6.5

### ISSUE THREAD

**Issue by tmcgilchrist** [vincenthz/hs-certificate#88]
Building version x509-store-1.6.4 with x509-1.6.5 fails with the following error:

```
Preprocessing library x509-store-1.6.4...
[1 of 3] Compiling Data.X509.Memory ( Data/X509/Memory.hs, dist/dist-sandbox-48d9b933/build/Data/X509/Memory.o )

Data/X509/Memory.hs:22:1: error:
    Failed to load interface for ‘Data.X509.EC’
    Perhaps you meant
      Data.X509.Ext
      Data.X509 (from x509-1.6.5)
      Data.X509.CRL
    Use -v to see a list of the files searched for.
[3 of 3] Compiling Data.X509.CertificateStore ( Data/X509/CertificateStore.hs, dist/dist-sandbox-48d9b933/build/Data/X509/CertificateStore.o )

Data/X509/CertificateStore.hs:13:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()

```

The version bounds on x509-store have a bound of `x509 >= 1.6` but in this instance it's referring to a file that only exists in x509-1.7.2

The second warning is nice to fix but not strictly necessary.

**Comment by vincenthz** [comment]
thanks for the report, fixed by x509-store 1.6.5. The import warning is for compat with older GHC that I have no plan to fix until I completely drop older GHC support (which is probably soon)

**Comment by tmcgilchrist** [comment]
Thanks 🏆
